pypeit/pypeit.py

```diff
@@ -670,44 +724,47 @@ def reduce_one(self, frames, det, bg_frames, std_outfile=None):
                             slits=True, clear=True)
 
         # Do it
-        skymodel, objmodel, ivarmodel, outmask, sobjs, scaleImg, waveImg, tilts = self.redux.run(
-            std_trace=std_trace, show_peaks=self.show,
-            ra=self.fitstbl["ra"][frames[0]], dec=self.fitstbl["dec"][frames[0]],
-            obstime=self.obstime)
-
-        # TODO -- Do this upstream
-        # Tack on detector and wavelength RMS
-        for sobj in sobjs:
-            sobj.DETECTOR = sciImg.detector
-            iwv = np.where(self.caliBrate.wv_calib.spat_ids == sobj.SLITID)[0][0]
-            sobj.WAVE_RMS =self.caliBrate.wv_calib.wv_fits[iwv].rms
-
-        # Construct table of spectral flexure
-        spec_flex_table = Table()
-        spec_flex_table['spat_id'] = self.caliBrate.slits.spat_id
-        spec_flex_table['sci_spec_flexure'] = self.redux.slitshift
-
-        # Construct the Spec2DObj
-        spec2DObj = spec2dobj.Spec2DObj(det=self.det,
-                                        sciimg=sciImg.image,
-                                        ivarraw=sciImg.ivar,
-                                        skymodel=skymodel,
-                                        objmodel=objmodel,
-                                        ivarmodel=ivarmodel,
-                                        scaleimg=scaleImg,
-                                        waveimg=waveImg,
-                                        bpmmask=outmask,
-                                        detector=sciImg.detector,
-                                        sci_spat_flexure=sciImg.spat_flexure,
-                                        sci_spec_flexure=spec_flex_table,
-                                        vel_corr=self.redux.vel_corr,
-                                        vel_type=self.par['calibrations']['wavelengths']['refframe'],
-                                        tilts=tilts,
-                                        slits=copy.deepcopy(self.caliBrate.slits))
-        spec2DObj.process_steps = sciImg.process_steps
-
-        # Return
-        return spec2DObj, sobjs
+        if objfind_run:
```

Comment:
Let's discuss this. I'm averse to doing this with an if statement, but understand why you've opted to do it this way. I'm rather concerned about the fact that we are rebuilding the sciimage twice. This was something I didn't anticipate. The CR rejection is very slow, so this is going to make all reductions a lot slower than I initially thought. I think we may want to write these files out to disk or something. Let's chat.